src/vivarium_cluster_tools/psimulate/paths.py

```diff
@@ -70,14 +72,29 @@ def from_entry_point_args(
         cls,
         *,
         command: str,
-        input_artifact_path: Optional[Path],
         result_directory: Path,
+        input_artifact_path: Optional[Path] = None,
+        input_model_spec_path: Path,
     ) -> "OutputPaths":
         launch_time = datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
 
         output_directory = result_directory
         if command == COMMANDS.run:
-            output_directory = output_directory / input_artifact_path.stem / launch_time
+            # Peek at the CLI args and model spec YAML to determine a good subdirectory name
+            # for the results. If we don't have an artifact, use the stem of the model spec,
+            # which must exist. NB: Actual determination of the artifact path is done in the
+            # model_specification.py code.
+            if input_artifact_path:
+                location = input_artifact_path.stem
+            else:
+                # check for YAML
+                with open(input_model_spec_path) as model_spec_file:
+                    model_spec = yaml.safe_load(model_spec_file)
+                try:
+                    location = Path(model_spec["configuration"]["input_data"]["artifact_path"]).stem
+                except KeyError:
```

Comment:
This kind of feels borderline like the type of error you shouldn't catch b/c it's expected behavior (please correct me if I'm wrong here). Regardless, another approach I defaulted to when thinking about this was implementing .get(). eg

```
location = model_spec.get('configuration', {}).get('input_data', {}).get('artifact_path')
if not location:
    location = input_model_spec_path.stem
```

Or I guess if you wanted to golf:
```
location = model_spec.get('configuration', {}).get('input_data', {}).get('artifact_path', input_model_spec_path.stem)
```

Reply:
I hesitated on the try/except because of that, it it fits so well. Do you think your code is clearer than the OP?

Reply:
I think it's clearer and results in one less indent. But I don't think it's a big deal either way

Reply:
OTOH, it doesn't seem like it's bad form in Python. EAFP.

Reply:
https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python

Reply:
The second comment on that thread actually points out the .get() approach. Seems like it doesn't matter much. I prefer minimizing indents (and .get() is readable to me), but what you have is fine as well.

Reply:
Looking at efficiency, the EAFP approach reads the dictionary once. The `.get()` approach reads it three times.